File: built-in plugins dsp/mb-restore-wind.cpp
/**
 * MB Wind Noise Filter
 * Category : effect
 * Type     : eq
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Adaptive wind noise detection and removal for outdoor recordings
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_RESTORE_WIND_H
#define MB_RESTORE_WIND_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbRestoreWind : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-restore-wind";
    static constexpr const char* PLUGIN_NAME    = "MB Wind Noise Filter";
    static constexpr const char* PLUGIN_TYPE    = "eq";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float sensitivity = 0.6f;  // range [0, 1]
    float reduction = 18f;  // range [0, 40]
    float lowCut = 80f;  // range [20, 200]
    };

    MbRestoreWind() = default;
    ~MbRestoreWind() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.sensitivity = std::clamp(params.sensitivity, 0f, 1f);
        params.reduction = std::clamp(params.reduction, 0f, 40f);
        params.lowCut = std::clamp(params.lowCut, 20f, 200f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Wind Noise Filter
        return input;
    }
};

#endif // MB_RESTORE_WIND_H

File: built-in plugins dsp/mb-restore-wind.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbRestoreWindPlugin: PluginDefinition = { id: 'mb-restore-wind', slug: 'mb-restore-wind', name: 'MB Wind Noise Filter', category: 'effect', type: 'eq' as any, version: '1.0.0', description: 'Adaptive wind noise detection and removal for outdoor recordings', author: 'Max Booster', grade: 'A', parameters: [{ id: 'sensitivity', name: 'Sensitivity', type: 'float', defaultValue: 0.6, minValue: 0, maxValue: 1, automatable: true }, { id: 'reduction', name: 'Reduction', type: 'float', defaultValue: 18, minValue: 0, maxValue: 40, automatable: true }, { id: 'lowCut', name: 'Low Cut', type: 'float', defaultValue: 80, minValue: 20, maxValue: 200, automatable: true }], defaultPreset: { sensitivity: 0.6, reduction: 18, lowCut: 80 } };

export default MbRestoreWindPlugin;
